### PR TITLE
Add support for cross-compiling for ppc64le platform

### PR DESCRIPTION
**Pull request by sumitd2**
..on the lines of https://github.com/grpc/grpc-java/pull/6441

### CODE REVIEW

File: buildscripts/kokoro/unix.sh [hunk 1]
```diff
@@ -90,6 +90,10 @@ if [[ -z "${ALL_ARTIFACTS:-}" ]]; then
     GRADLE_FLAGS+=" -x grpc-compiler:generateTestProto -x grpc-compiler:generateTestLiteProto"
     GRADLE_FLAGS+=" -x grpc-compiler:testGolden -x grpc-compiler:testLiteGolden"
     GRADLE_FLAGS+=" -x grpc-compiler:testDeprecatedGolden -x grpc-compiler:testDeprecatedLiteGolden"
+  elif [[ $ARCH == "ppcle_64" ]]; then
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh:93]: Let's not duplicate the lines within. They are all the same with aarch_64. Instead, combine this if with the one above and check for aarch_64 or (`||`) ppcle_64.

File: compiler/check-artifact.sh [hunk 2]
```diff
@@ -108,6 +110,8 @@ checkDependencies ()
     elif [[ "$ARCH" == aarch_64 ]]; then
       dump_cmd='aarch64-linux-gnu-objdump -x '"$1"' |grep "NEEDED"'
       white_list="linux-vdso\.so\.1\|libpthread\.so\.0\|libm\.so\.6\|libc\.so\.6\|ld-linux-aarch64\.so\.1"
+    elif [[ "$ARCH" == ppcle_64 ]]; then
+      white_list="linux-vdso64\.so\.1\|libpthread\.so\.0\|libm\.so\.6\|libc\.so\.6\|ld64\.so\.2\|libstdc++\.so\.6\|libgcc_s\.so\.1"
```
Comment (ejona86) [inline compiler/check-artifact.sh:114]: I don't think libstdc and libgcc_s should be in this list. If those show up, then the static linking isn't working as we'd hope.

I don't think the x86 ldd can process the ppc binary here. It seems dump_cmd would need to be defined.

**Comment by sumitd2** [comment]
Not sure how to fix this for cross compilation:
```
> Task :grpc-compiler:linkJava_pluginExecutable FAILED
/usr/lib/gcc-cross/powerpc64-linux-gnu/5/../../../../powerpc64-linux-gnu/bin/ld: skipping incompatible /tmp/protobuf/lib/libprotoc.a when searching for -lprotoc
/usr/lib/gcc-cross/powerpc64-linux-gnu/5/../../../../powerpc64-linux-gnu/bin/ld: skipping incompatible //usr/local/lib/libprotoc.a when searching for -lprotoc
/usr/lib/gcc-cross/powerpc64-linux-gnu/5/../../../../powerpc64-linux-gnu/bin/ld: cannot find -lprotoc
/usr/lib/gcc-cross/powerpc64-linux-gnu/5/../../../../powerpc64-linux-gnu/bin/ld: skipping incompatible /tmp/protobuf/lib/libprotobuf.a when searching for -lprotobuf
/usr/lib/gcc-cross/powerpc64-linux-gnu/5/../../../../powerpc64-linux-gnu/bin/ld: skipping incompatible //usr/local/lib/libprotobuf.a when searching for -lprotobuf
/usr/lib/gcc-cross/powerpc64-linux-gnu/5/../../../../powerpc64-linux-gnu/bin/ld: cannot find -lprotobuf
collect2: error: ld returned 1 exit status

```

**Comment by sumitd2** [comment]
The host string looks wrong to me. Adding the fix...

```
+ ./configure --disable-shared --host=ppc64le-linux-gnu --prefix=/tmp/protobuf-cache/3.19.2/Linux-ppcle_64
checking host system type... powerpc64le-unknown-linux-gnu
checking target system type... powerpc64le-unknown-linux-gnu
checking for ppc64le-linux-gnu-gcc... no
checking for gcc... gcc
**checking whether we are cross compiling... no**
checking for ppc64le-linux-gnu-g++... no
checking for g++... g++

+ ./configure --disable-shared --host=aarch64-linux-gnu --prefix=/tmp/protobuf-cache/3.19.2/Linux-aarch_64
checking build system type... x86_64-pc-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking target system type... aarch64-unknown-linux-gnu
checking for aarch64-linux-gnu-gcc... aarch64-linux-gnu-gcc
**checking whether we are cross compiling... yes**
```

**Comment by sumitd2** [comment]
exe was built successfully. I think we are using the wrong _objdump_?

```
> Task :grpc-compiler:checkArtifacts FAILED

Checking format of build/artifacts/java_plugin/protoc-gen-grpc-java.exe
Format=elf64-little
Assertion failed:  "elf64-little" == "elf64-powerpcle"
File "check-artifact.sh", line 65

FAILURE: Build failed with an exception.
```

File: buildscripts/kokoro/unix.sh [hunk 3]
```diff
@@ -86,7 +88,7 @@ fi
 LOCAL_MVN_TEMP=$(mktemp -d)
 # Note that this disables parallel=true from GRADLE_FLAGS
 if [[ -z "${ALL_ARTIFACTS:-}" ]]; then
-  if [[ $ARCH == "aarch_64" ]]; then
+  if [ "$ARCH" = "aarch_64" ] || [ "$ARCH" = "ppcle_64" ]; then
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh:91]: We prefer `[[` over `[`. This is best written as `[[ "$ARCH" = "aarch_64" || "$ARCH" = "ppcle_64" ]]`

**Review by ejona86 — APPROVED** [review]
The change in check-artifacts for getting the format should probably be cleaned up in the future, but we can do that later.

**Review by sanjaypujare — APPROVED** [review]
LGTM

**Pull request merged**